### PR TITLE
IOS-7725 Load icons asynchronously

### PR DESCRIPTION
**Pull request by amegias**
https://jira.tid.es/browse/IOS-7725

The icons are being loaded in a sync flow. It blocks the view creation and then the presentation (See the ticket->video).

I've moved to an async flow to load the icons and then, fix this ticket.

We will work on placeholders, error cases, etc in the future

https://user-images.githubusercontent.com/9945756/192541308-d084e469-4e76-4a26-911c-47769e00c97b.mov

### CODE REVIEW

File: Sources/Mistica/Components/Sheet/View/SheetView.swift [hunk 1]
```diff
@@ -135,17 +135,7 @@ private extension SheetView {
...
-                       let data = try? Data(contentsOf: url) {
-                        let lightImage = UIImage(data: data)
-                        if let iconDark = icon.urlDark,
-                           let urlDark = URL(string: iconDark),
-                           let dataDark = try? Data(contentsOf: urlDark),
-                           let darkImage = UIImage(data: dataDark) {
-                            lightImage?.imageAsset?.register(darkImage, with: .init(userInterfaceStyle: .dark))
-                        }
-                        imageView.image = lightImage
-                    }
```
Comment (amegias) [inline Sources/Mistica/Components/Sheet/View/SheetView.swift:148]: moved to an extension and fix the flow with an async strategy

File: Sources/Mistica/Components/Sheet/View/SheetView.swift [hunk 2]
```diff
@@ -135,17 +135,7 @@ private extension SheetView {
                     imageView.heightAnchor.constraint(equalToConstant: icon.size.value).isActive = true
                     imageView.widthAnchor.constraint(equalToConstant: icon.size.value).isActive = true
                     imageView.contentMode = .scaleAspectFit
-                    if let url = URL(string: icon.url),
-                       let data = try? Data(contentsOf: url) {
-                        let lightImage = UIImage(data: data)
-                        if let iconDark = icon.urlDark,
-                           let urlDark = URL(string: iconDark),
-                           let dataDark = try? Data(contentsOf: urlDark),
```
Comment (amegias) [inline Sources/Mistica/Components/Sheet/View/SheetView.swift:143]: this line is blocking the main thread

File: Sources/Mistica/Utils/Extensions/UIImageView+URL.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import UIKit
+
+extension UIImageView {
+    /// Loads the urls asynchronously
+    func load(url: URL, urlForDarkMode: URL? = nil) {
+        DispatchQueue.global().async { [weak self] in
+            guard let imageData = try? Data(contentsOf: url) else { return }
+            if let urlForDarkMode = urlForDarkMode,
+               let darkImageData = try? Data(contentsOf: urlForDarkMode) {
```
Comment (amegias) [inline Sources/Mistica/Utils/Extensions/UIImageView+URL.swift:17]: load this in background thread

File: Sources/Mistica/Utils/Extensions/UIImageView+URL.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                self?.updateImage(imageData: imageData, darkImageData: darkImageData)
+            } else {
+                self?.updateImage(imageData: imageData)
```
Comment (amegias) [inline Sources/Mistica/Utils/Extensions/UIImageView+URL.swift:20]: update it in the MThread

**Review by jmpg93 — APPROVED** [review]

**Pull request merged**